tfx/src/main/scala/com/spotify/noether/tfx/TfmaConverter.scala

```diff
@@ -0,0 +1,300 @@
+                                                   Map[Double, (Double, Double, Long)],
+                                                   CalibrationHistogram] =
+    new TfmaConverter[Prediction[Double, Double],
+                      Map[Double, (Double, Double, Long)],
+                      CalibrationHistogram] {
+      override def convertToTfmaProto(underlying: CalibrationHistogram)
+        : Aggregator[Prediction[Double, Double], Map[Double, (Double, Double, Long)], EvalResult] =
+        underlying.andThenPresent { calibrationHistogram =>
+          val plot = buildCalibrationHistogramPlot(calibrationHistogram)
+          EvalResult(Plot.CalibrationHistogram(plot))
+        }
+    }
+
+  private def denseMatrixToMetric(threshold: Option[Double] = None)(
+    matrix: DenseMatrix[Long]): MetricsForSlice = {
+    val cm = denseMatrixToConfusionMatrix(threshold)(matrix)
+    MetricsForSlice
+      .newBuilder()
+      .setSliceKey(SliceKey.getDefaultInstance)
+      .putMetrics(
+        "Noether_ConfusionMatrix",
+        MetricValue
+          .newBuilder()
+          .setConfusionMatrixAtThresholds(cm)
+          .build()
+      )
+      .build()
+  }
+
+  private def confusionMatrixToMetric(cm: ConfusionMatrixAtThresholds): MetricsForSlice = {
+    MetricsForSlice
+      .newBuilder()
+      .setSliceKey(SliceKey.getDefaultInstance)
+      .putMetrics(
+        "Noether_ConfusionMatrix",
+        MetricValue
+          .newBuilder()
+          .setConfusionMatrixAtThresholds(cm)
+          .build()
+      )
+      .build()
+  }
+
+  private def denseMatrixToConfusionMatrix(threshold: Option[Double] = None)(
+    matrix: DenseMatrix[Long]): ConfusionMatrixAtThresholds = {
+    val tp = matrix.valueAt(1, 1).toDouble
+    val tn = matrix.valueAt(0, 0).toDouble
+    val fp = matrix.valueAt(1, 0).toDouble
+    val fn = matrix.valueAt(0, 1).toDouble
+
+    val cmBuilder = ConfusionMatrixAtThreshold
+      .newBuilder()
+      .setFalseNegatives(fn)
+      .setFalsePositives(fp)
+      .setTrueNegatives(tn)
+      .setTruePositives(tp)
+      .setPrecision(tp / (tp + fp))
+      .setRecall(tp / (tp + fn))
+
+    threshold.foreach(cmBuilder.setThreshold)
+    ConfusionMatrixAtThresholds
+      .newBuilder()
+      .addMatrices(cmBuilder.build())
+      .build()
+  }
+
+  private def buildDoubleMetric(name: String, value: Double): MetricsForSlice =
+    MetricsForSlice
+      .newBuilder()
+      .setSliceKey(SliceKey.getDefaultInstance)
+      .putMetrics(name,
+                  MetricValue
+                    .newBuilder()
+                    .setDoubleValue(DoubleValue.newBuilder().setValue(value))
+                    .build())
+      .build()
+
+  private def buildDoubleMetrics(metrics: Map[String, Double]): MetricsForSlice = {
+    MetricsForSlice
+      .newBuilder()
+      .setSliceKey(SliceKey.getDefaultInstance)
+      .putAllMetrics(metrics.mapValues { m =>
```

Comment:
🎨 move this outside; more readable?

Reply:
Is this what you mean by move this outside?
```
val metrics = metrics.mapValues { m =>
        MetricValue
          .newBuilder()
          .setDoubleValue(
            DoubleValue
              .newBuilder()
              .setValue(m))
          .build()
      }.asJava

    MetricsForSlice
      .newBuilder()
      .setSliceKey(SliceKey.getDefaultInstance)
      .putAllMetrics(metrics)
      .build()
```